lib/nats.js

```diff
@@ -797,6 +801,23 @@ Client.prototype.processInbound = function() {
 	if (client.checkTLSMismatch() === true) {
 	  return;
 	}
+
+  // Always try to read the connect_urls from info
+  if(client.info.connect_urls && client.info.connect_urls.length > 0) {
+    // don't add duplicates
+    var known = [];
+    client.servers.forEach(function(server) {
+      known.push(server.url.href);
+    });
+    // add new ones
+    client.info.connect_urls.forEach(function(server) {
+      var u = 'nats://' + server;
+      if(known.indexOf(u) === -1) {
+        client.addServer(u);
```

Comment:
Just to be clear, we don't want to randomize the pool when we add things after the initial connect. We can randomize the received URLs *before* adding to the pool, but not randomize the entire pool after initial connect.